ddtrace/contrib/internal/botocore/services/stepfunctions.py

```diff
@@ -34,8 +36,15 @@ def update_stepfunction_input(ctx: core.ExecutionContext, params: Any) -> None:
         return
 
     input_obj["_datadog"] = {}
-
-    core.dispatch("botocore.stepfunctions.update_input", [ctx, None, None, input_obj["_datadog"], None])
+    results = core.dispatch_with_results("botocore.stepfunctions.update_input", [ctx, None, None, input_obj, None])
```

Comment:
Instead of using `dispatch_with_results`, I recommend using `ExecutionContext.set_item` and `ExecutionContext.get_item` to pass information between the event handler and the dispatching code. `dispatch_with_results` has problems and is going to be deprecated soon.

Reply:
Good call. Updated.